enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,475 @@
+last-updated: 2021-06-07
+status: implementable
+---
+
+# ExternalDNS Operator
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Open Questions](#open-questions)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade and Downgrade Strategy](#upgradedowngrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposal is for adding an operator to manage
+[ExternalDNS](https://github.com/kubernetes-sigs/external-dns). ExternalDNS has been chosen for managing external DNS
+requirements of OpenShift clusters. Initially, the operator will focus on managing external DNS records for OpenShift
+[Routes](https://github.com/openshift/api/blob/master/route/v1/types.go), in addition to Kubernetes Services and
+instances of a DNSRecord custom resource. Additionally, the operator will have its own CRD that will initially expose limited
+configuration capabilities. Existing operators will be used to guide the design of the ExternalDNS Operator. Tooling will be
+included to make it easy to build, run, test, etc. the operator for OpenShift, and for vanilla Kubernetes.
+
+## Motivation
+
+The [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) is generally accepted, and the
+benefits of operators to OpenShift are well-known.  This enhancement proposal aims to bring the benefits of operators to
+ExternalDNS. ExternalDNS is a controller that manages external DNS records. As with other platform components,
+ExternalDNS requires an operator to provide lifecycle management of the component.
+
+The ExternalDNS upstream community would also theoretically benefit from the introduction of an ExternalDNS operator as an
+ExternalDNS operator compatible with vanilla kubernetes (in addition to OpenShift) would presumably be utilized by other downstream
+ExternalDNS users to simplify the deployment and lifecycle management of ExternalDNS. Ideally, the upstream ExternalDNS community
+would ultimately participate in the design and construction of the ExternalDNS Operator.
+
+### Goals
```

Comment:
I would love to see each of these goals appear as subsections in the implementation details, where the goal is mentioned.

Reply:
Some of these goals are straight forward. Do you think the implementation details section is missing anything in particular?